tests/integration_tests/functional/test_snapshot_basic.py

```diff
@@ -37,81 +34,65 @@ def _get_guest_drive_size(ssh_connection, guest_dev_name="/dev/vdb"):
 # - Microvm: 2vCPU with 512 MB RAM
 # TODO: Multiple microvm sizes must be tested in the async pipeline.
-@pytest.mark.parametrize(
-    "microvm", ARTIFACTS.microvms(keyword="2vcpu_512mb"), ids=lambda x: x.name()
-)
-@pytest.mark.parametrize("kernel", ARTIFACTS.kernels(), ids=lambda x: x.name())
-@pytest.mark.parametrize(
-    "disk", ARTIFACTS.disks(keyword="ubuntu"), ids=lambda x: x.name()
-)
 @pytest.mark.parametrize("snapshot_type", [SnapshotType.DIFF, SnapshotType.FULL])
 def test_5_snapshots(
     bin_cloner_path,
     bin_vsock_path,
-    test_fc_session_root_path,
-    microvm,
-    kernel,
-    disk,
+    tmp_path,
+    microvm_factory,
+    guest_kernel,
+    rootfs,
     snapshot_type,
+    network_config,
 ):
     """
     Create and load 5 snapshots.
 
     @type: functional
     """
     logger = logging.getLogger("snapshot_sequence")
-
     vm_builder = MicrovmBuilder(bin_cloner_path)
     seq_len = 5
     diff_snapshots = snapshot_type == SnapshotType.DIFF
 
-    disk.download()
-    kernel.download()
-    microvm.download()
-
-    # Create a rw copy artifact.
-    root_disk = disk.copy()
-    # Get ssh key from read-only artifact.
-    ssh_key = disk.ssh_key()
-    # Create a fresh microvm from artifacts.
-    vm_instance = vm_builder.build(
-        kernel=kernel,
-        disks=[root_disk],
-        ssh_key=ssh_key,
-        config=microvm,
-        diff_snapshots=diff_snapshots,
+    vm = microvm_factory.build(guest_kernel, rootfs)
+    vm.spawn()
+    vm.basic_config(
+        vcpu_count=2,
+        mem_size_mib=512,
+        track_dirty_pages=diff_snapshots,
     )
-    basevm = vm_instance.vm
-    basevm.vsock.put(vsock_id="vsock0", guest_cid=3, uds_path=f"/{VSOCK_UDS_PATH}")
-
-    basevm.start()
+    tap, host_ip, guest_ip = vm.ssh_network_config(network_config, "1")
+    vm.vsock.put(vsock_id="vsock0", guest_cid=3, uds_path=VSOCK_UDS_PATH)
+    vm.start()
     # Verify if guest can run commands.
-    exit_code, _, _ = basevm.ssh.run("sync")
+    exit_code, _, _ = vm.ssh.run("sync")
     assert exit_code == 0
 
     vm_blob_path = "/tmp/vsock/test.blob"
     # Generate a random data file for vsock.
-    blob_path, blob_hash = make_blob(test_fc_session_root_path)
+    blob_path, blob_hash = make_blob(tmp_path)
     # Copy the data file and a vsock helper to the guest.
-    _copy_vsock_data_to_guest(basevm.ssh, blob_path, vm_blob_path, bin_vsock_path)
+    _copy_vsock_data_to_guest(vm.ssh, blob_path, vm_blob_path, bin_vsock_path)
 
     logger.info("Create %s #0.", snapshot_type)
     # Create a snapshot builder from a microvm.
-    snapshot_builder = SnapshotBuilder(basevm)
+    snapshot_builder = SnapshotBuilder(vm)
+    iface = NetIfaceConfig(host_ip, guest_ip, tap.name, "eth1", 30)
 
     # Create base snapshot.
-    snapshot = snapshot_builder.create([root_disk.local_path()], ssh_key, snapshot_type)
-
+    ssh_key = rootfs.ssh_key()
+    disks = [vm.rootfs_file]
+    snapshot = snapshot_builder.create(
```

Comment:
It isn't clear clear what the difference here is between using the `SnapshotBuilder` and `SnapshotHelper` types, and if they're related to dropping the TestMatrix code, or if it is for another reason. Is a custom network interface required?

Reply:
I don't follow the comment abt `SnapshotHelper`. It doesn't appear here at all?

The reason to provide a network interface to the snapshot builder is because the microvm class does not know which network interfaces it has. This is something that should be cleaned up in the future, so that the client does not have to pass them around.

Reply:
@pb8o I saw that in the next commit you add to the `Microvm` knowledge about its network interfaces. Could we re-use that here and avoid passing `net_ifaces` to the snapshot builder?

Reply:
> @pb8o I saw that in the next commit you add to the Microvm knowledge about its network interfaces. Could we re-use that here and avoid passing net_ifaces to the snapshot builder?

That's my plan, yeah :) Microvm should have all the information needed so the client does not have to pass interfaces or ssh keys. I will do that in a follow-up, as this PR is already long.